services/kaze/controllers/application.go

```diff
@@ -228,29 +230,27 @@ func TransferApplicationOwnership(c *gin.Context) {
 
 // FetchMetrics retrieves the metrics of an application's container
 func FetchMetrics(c *gin.Context) {
+	c.Writer.Header().Set("Content-Type", "text/event-stream")
 	appName := c.Param("app")
-	filter := utils.QueryToFilter(c.Request.URL.Query())
-	var timeSpan int64
-
-	for unit, converter := range timeConversionMap {
-		if val, ok := filter[unit].(string); ok {
-			timeVal, err := strconv.ParseInt(val, 10, 64)
-			if err != nil {
-				continue
-			}
-			timeSpan += timeVal * converter
-		}
-	}
+	metricsFetchInterval := c.Query("fetch_interval")
+	metricsCount := c.Query("fetch_count")
 
-	metrics := mongo.FetchContainerMetrics(types.M{
-		mongo.NameKey: appName,
-		mongo.TimestampKey: types.M{
-			"$gte": time.Now().Unix() - timeSpan,
-		},
-	})
+	count, _ := strconv.ParseInt(metricsCount, 10, 64)
 
-	c.JSON(200, gin.H{
-		"success": true,
-		"data":    metrics,
+	c.Stream(func(w io.Writer) bool {
+		metrics := mongo.FetchContainerMetrics(types.M{
+			mongo.NameKey: appName,
+			mongo.TimestampKey: types.M{
+				"$gte": time.Now().Unix() - int64(configs.ServiceConfig.Mizu.MetricsInterval*time.Second),
+			},
+		}, count)
+		c.SSEvent("metrics", metrics)
+		fetchInt, _ := strconv.ParseInt(metricsFetchInterval, 10, 64)
```

Comment:
Check for error handling here too in case "metricsFetchInterval" is a string like "meow"